src/R/Components/Impl/PackageManager/Implementation/RPackageManager.cs

```diff
@@ -180,8 +229,40 @@ internal class RPackageManager : IRPackageManager {
             }
         }
 
+        private async Task CreatePackageQuerySessionAsync() {
+            await _pkgQuerySessionSemaphore.WaitAsync();
```

Comment:
Session can't dispose itself, you don't need to subscribe for `_pkgQuerySession.Disposed` if you don't share its instance